ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -425,6 +428,51 @@ CloudBlobMetadata getMetadataOrNull(BlobId blobId) throws DocumentClientExceptio
     }
   }
 
+  /**
+   * Get the list of blobs in the specified partition that belong to the specified container.
+   * @param partitionPath the partition to query.
+   * @param accountId account id of the container.
+   * @param containerId container id of the container.
+   * @param queryLimit max number of blobs to return.
+   * @return a List of {@link CloudBlobMetadata} referencing the blobs belonging to the deprecated containers.
+   * @throws DocumentClientException in case of any error.
+   */
+  List<CloudBlobMetadata> getContainerBlobs(String partitionPath, short accountId, short containerId, int queryLimit)
+      throws DocumentClientException {
+    String query = String.format(CONTAINER_BLOBS_QUERY, accountId, containerId);
+    SqlQuerySpec querySpec = new SqlQuerySpec(query);
+    FeedOptions feedOptions = new FeedOptions();
```

Comment:
The code from here down is nearly identical to getDeadBlobs.  Please refactor into a common class that take SqlQuerySpec as argument.

Reply:
Also, since we aren't time bucketing the queries like dead blob compaction does, we could run into the same throttling issues that prompted the use of time bucketing there.